File: cpp/Sorted-Merge-Lcci/Sorted-Merge-Lcci.cpp
class Solution {
public:
    void merge(vector<int>& A, int m, vector<int>& B, int n) {
        int i = m-1;
        int j = n-1;
        int index =  m+n-1;
        while (i>=0 && j>=0)
        {
            if(A[i]>=B[j])
            {
                A[index--] = A[i--];
            }
            else
            {
                A[index--] = B[j--];
            }
        }
        // 下面这个while没必要写，这时候index 和i是相等的。
        // while (i>=0)  A[index--] = A[i--];
        while (j>=0)  A[index--] = B[j--];
    }
};

File: python/Sorted-Merge-Lcci/Sorted-Merge-Lcci.py
class Solution:
    def merge(self, A: List[int], m: int, B: List[int], n: int) -> None:
        """
        Do not return anything, modify A in-place instead.
        """
        i = m-1
        j = n-1
        index = m+n-1
        while i>=0 and j>=0:
            if A[i] >= B[j]:
                A[index] = A[i]
                index-=1
                i-=1
            else:
                A[index] = B[j]
                index -=1
                j-=1
        # 比较完B还有剩下的，全填到A前面即可
        A[:j+1] = B[:j+1]
